src/model.py

```diff
@@ -2048,22 +2048,21 @@ def init_components(self, subtasks):
         relations = torch.nn.ModuleDict({})
         attn_linears = torch.nn.ModuleDict({})
         for stack in range(stack_count):
+            relations[f"stack_{stack}"] = torch.nn.ModuleDict({})
             for rel in attention_relations:
-                relations[f"stack_{stack}"] = torch.nn.ModuleDict({
-                    f"{rel[0]}_at_{rel[1]}": torch.nn.ModuleDict({
-                        "attn": self.attn_block(cnn_dim),
+                relations[f"stack_{stack}"][f"{rel[0]}_at_{rel[1]}"] = torch.nn.ModuleDict({
+                    "attn": self.attn_block(cnn_dim),
```

Comment:
This was previously causing all attentions relations for a given stack except for the last to be overwritten. Now, the stack dict is created first, then relations are added as looped through